### PR TITLE
Mark privacyidea token in the qr code

### LINKED ISSUE #3902: Mark privacyidea token in the qr code
We want to be able to distinguish privacyidea and non-privacyidea token in the app, so we are able to offer functionality based on that info. We need some kind of extra in the qr code for that.

### PR DESCRIPTION
**Pull request by jona-samuel**
closes #3902 

Is 'creator' fine as the new tag?

frame

This is an example QR code. Please make sure that all apps work with it.

### CODE REVIEW

**Comment by frankmer** [comment]
privacyIDEA Authenticator can handle hotp or totp in upper case. To work with any common authenticator it has to be lowercase.
The authenticators I used just ignore the creator parameter and can scan the QR if I set the token type to lower case.

I will implement our app to use the creator parameter.

File: tests/test_lib_apps.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_01_apps_urls(self):
         r = create_google_authenticator_url(tokentype="TOTP", key="123456",
                                             period=60)
         self.assertTrue("otpauth://TOTP/mylabel?secret=CI2FM&period"
```
Comment (frankmer) [inline tests/test_lib_apps.py:22]: totp must be lowercase

**Review by frankmer — CHANGES_REQUESTED** [review]

File: tests/test_lib_apps.py [hunk 2]
```diff
@@ -17,10 +17,10 @@ def test_01_apps_urls(self):
         self.assertTrue("otpauth://hotp/mylabel?secret=CI2FM6A&counter=1" in
                         r, r)
 
-        r = create_google_authenticator_url(tokentype="TOTP", key="123456",
+        r = create_google_authenticator_url(tokentype="totp", key="123456",
```
[the remark this answers is not in the document]
Comment (jona-samuel) [reply tests/test_lib_apps.py]: [14084c2](https://github.com/privacyidea/privacyidea/pull/3909/commits/14084c252925b2247c5a4415a41506c4837c404a)

File: privacyidea/lib/apps.py [hunk 3]
```diff
@@ -162,6 +162,7 @@ def create_google_authenticator_url(key=None, user=None,
     return ("otpauth://{tokentype!s}/{label!s}?secret={secret!s}&"
             "{counter!s}{hash!s}{period!s}"
             "digits={digits!s}&"
+            "creator=privacyidea&"
```
[the remark this answers is not in the document]
Comment (jona-samuel) [reply privacyidea/lib/apps.py]: [14084c2](https://github.com/privacyidea/privacyidea/pull/3909/commits/14084c252925b2247c5a4415a41506c4837c404a)

File: privacyidea/lib/apps.py [hunk 4]
```diff
@@ -153,6 +153,7 @@ def create_google_authenticator_url(key=None, user=None,
         hash_algo = ""
 
     if tokentype.lower() == "totp":
+        tokentype = "totp"
```
Comment (plettich) [inline privacyidea/lib/apps.py:156]: I meant something like
`tokentype = tokentype.lower()`
at the beginning of the function. So what ever is passed to the function as a tokentype is converted to lower case.

**Review by plettich — CHANGES_REQUESTED** [review]

**Review by plettich — APPROVED** [review]

**Pull request merged**